helpers/preview/bin/preview-rst

```diff
@@ -0,0 +1,313 @@
+
+
+# =============================================================================
+# Preview an RST file
+# =============================================================================
+
+# This script can be used to generate an HTML preview of any RST file you may
+# be editing.
+#
+# Usage:
+#
+#    preview-rst <RST_FILE>
+#
+# This script is not intended to be run by hand. It has been written with the
+# purpose of being integrated into a desktop editing environment. Integration
+# is possible if you are using an editor that allows you to extend its
+# functionality with arbitrary scripting.
+#
+# In particular, you may get the most use out of this script if you bind it to
+# a keyboard shortcut like <Command-P> (for "preview"). This would likely
+# involve overriding the "Print" keyboard shortcut of your editor, which,
+# depending on your situation, may not be too important.
+#
+# This script has two modalities:
+#
+# - Running this script on RST files in a Sphinx project should open a new
+#   browser window to the corresponding HTML page.
+#
+#   If you are editing lots of files, this can save a significant amount of
+#   time because it does not require you to navigate to each page manually to
+#   check your work.
+#
+# - Running this script on a standalone file (e.g., `CHANGES.rst`) will build
+#   an HTML version and open a new browser window to display it.
+#
+#   This is helpful because without pushing a file to GitHub (or elsewhere)
+#   there is no easy way to preview your work. Indeed, lots of errors may creep
+#   in because of this limitation.
+#
+# For standalone files, there are two strategies currently implemented:
+#
+# - The basic strategy (`display_standalone`) uses pandoc to generate a simple
+#   HTML file. This file has no styling whatsoever and, as a result, the result
+#   may be hard to read. This strategy has been disabled in favor of the Sphinx
+#   strategy (see next item).
+#
+# - The Sphinx strategy (`display_standalone_sphinx`) creates a temporary
+#   Sphinx project for your RST file, starts `sphinx-autobuild`, and opens a
+#   new browser window.
+#
+#   This strategy has many advantages:
+#
+#   - At the moment, the Read The Docs theme is used. This styles your RST in
+#     an easy-to-ready manner.
+#
+#   - The `sphinx-autobuild` program features live reloading, meaning that when
+#     you edit the RST file, the HTML will be rebuilt, and the browser window
+#     will automatically refresh to show your changes.
+
+
+# Caution
+# -----------------------------------------------------------------------------
+
+# This script is a work in progress and features may be added, removed, or
+# changed significantly.
+#
+# TODO:
+#
+# - Assorted `TODO` markers scattered throughout the code
+#
+# - Add wrappers for specific editors (e.g., BBEdit)
+#
+# - Needs to exclude some directives (e.g., admonitions) when generating
+#   standalone files to mimic GitHub's limited RST support
+
+
```

Comment:
Another TODO item from my mind:

- Add capability to instantly render whole directories instead of single files only. When detecting an `index.rst` within the directory, open that file in the Browser.

Reply:
great suggestion! can you create an issue please?

Reply:
created a feature request: https://github.com/crate/crate-docs/issues/79